nexus/social_media/services.py

```diff
@@ -30,5 +77,32 @@ def post_to_facebook(post_id):
             page_graph.put_photo(image=post.image.file.open('rb'))
     elif post.text:
         page_graph.put_object(
-            parent_object=settings.FB_PAGE_ID, connection_name='feed',  message=post.text
+            parent_object=settings.FB_PAGE_ID, connection_name='feed', message=post.text
+        )
+
+
+def publish_on_social_media():
+    if settings.LIMIT_POSTS is True and int(settings.MAX_POSTS_AT_ONCE) > 0:
+        posts = Post.objects.filter(
+            is_approved=True, is_posted=False, scheduled_time__lte=datetime.now()
+        )[:int(settings.MAX_POSTS_AT_ONCE)]
+    else:
+        posts = Post.objects.filter(
+            is_approved=True, is_posted=False, scheduled_time__lte=datetime.now()
         )
+
+    # Before bulk update, saving the IDs of posts along with there publishing platforms.
+    post_platform = {}
+    for post in posts:
+        post_platform.update({post.id: post.posted_at})
+
+    if settings.LIMIT_POSTS is True and int(settings.MAX_POSTS_AT_ONCE) > 0:
+        Post.objects.filter(id__in=posts).update(is_posted=True, posted_time=timezone.now())
```

Comment:
Why not just use `posts` directly here?

also, why not club this and the part above. Both blocks are using the same condition.

Reply:
In reference to the above if-else condition:
1.If the first `if` condition is true then `posts` become a __sliced__ queryset. And it is inbuild restriction of django that we can not update a sliced queryset simply with `posts.update()`. That's why I had to use this.
2. If the first `if` condition is false, then `else` part does *not* make `posts` as sliced queryset and I could directly use `posts.update()`

Which means, using `posts.update()` is not valid for both conditions. But just now I tested that using `Post.objects.filter().update()` is valid for both conditions. So I am removing the second `if-else` condition altogether.